### PR TITLE
Logging updates

### PR DESCRIPTION
**Pull request by puckybreg**
If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### What's this PR do?
Updates Winston & adds swagger-stats to the API
#### Related JIRA tickets:
https://jira.amida-tech.com/browse/ORANGE-625
#### Screenshots (if appropriate):

### CODE REVIEW

**Comment by orndorffgrant** [comment]
It looks like there is another logger enabled in `config/express.js:23` called `morgan`. Should we remove this so that we only have nice winston json logging?

Otherwise the winston logging and swagger stats worked for me!

**Comment by puckybreg** [comment]
@orndorffgrant Yeah, it's interesting, both `winston` & `morgan` existed in the project before I did these updates. My best guess is that they are used for different types of logging? At least in the console when running this service I don't see duplicate logging.

**Comment by orndorffgrant** [comment]
When I run `yarn start` I am seeing double logging
screen shot 2018-09-19 at 11 16 32 am
(Notice the json logging for the 401s and then the prettier human readable colored 401 messages below them)

When I comment out `morgan` I don't get the prettier lines of logging.

So I think we should either get rid of `morgan`, or keep `morgan` on during development and only turn `winston` on for production

**Comment by orndorffgrant** [comment]
Also, (sorry I keep noticing more things), it looks like there is already an actuator in there. See `config/express.js:61`

If you hit `/actuator/metrics` you get metrics for this service. This is redundant now that we have swagger stats right? So we should probably strip out this actuator as well?

**Comment by puckybreg** [comment]
This PR has been updated

**Comment by orndorffgrant** [comment]
There are some actuator tests in server/tests/misc.spec.js that could be modified to test the new swagger endpoints in a basic fashion if you're up to it, or we can just delete them for now.

**Review by orndorffgrant — APPROVED** [review]
👍

**Comment by puckybreg** [comment]
oof, you got me again @orndorffgrant. I think I found a compromise between fixing and removing the tests. Swagger-Stats doesn't have a endpoint just for info on the app (or at least not one that is obvious on the Swagger-Stats docs).

File: index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import config from './config/config';
 import app from './config/express';
 /* eslint-disable no-unused-vars */
 import db from './config/sequelize';
+import logger from './config/winston';
 
 const debug = require('debug')('amida-api-boilerplate:index');
```
Comment (rmharrison) [inline index.js]: ## Immediate recommendation (blocking PR)
- Disable debug logging in production
- Right now, debug logging (e.g. postgres), happens for both production and development environments to stdout. 
- Why: All logging in stdout (at least in prod) should be JSON structured

Confirm with
```
$ work@Hargrove:~/git/amida-tech/amida-messaging-microservice (logging-updates)$ NODE_ENV='production' yarn start
...
Executing (default): CREATE TABLE IF NOT EXISTS "Users" ("id"  SERIAL , "username" VARCHAR(255) NOT NULL, "createdAt" TIMESTAMP WITH TIME ZONE NOT NULL, "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL, PRIMARY KEY ("id"));
Executing (default): SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND t.relkind = 'r' and t.relname = 'Users' GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;
```

## Next step (not blocking PR)
May want to discuss what to do with non-JSON structured debug logging in development.
While turning-off in prod is within the scope of this PR, there could be some debate about what to do in dev.
I'm inclined towards **always JSON**, with folks separating the prettification-concern, but the community seems split. From what I can tell, Winston folks bake prettificiation into the application for debug, while bunyan folks prettify seperately (all JSON in both prod and dev).

If I ruled the world:
- Remove debug entirely and use the same logger (winston)
- Separate logger name for postgres
- All JSON

YMMV.

File: index.js [hunk 2]
```diff
@@ -14,7 +15,10 @@ function startServer() {
     if (!module.parent) {
     // listen on port config.port
         app.listen(config.port, () => {
-            debug(`server started on port ${config.port} (${config.env})`);
+            logger.info({
+                service: 'messaging-service',
+                message: `server started on port ${config.port} (${config.env})`,
```
Comment (rmharrison) [inline index.js]: **Non-blocking**
`config.port` and `config.env` are perfect candidates for log keys. Agnostic about punting to https://jira.amida.com/browse/ORANGE-795 when the log format will be updated anyways.

File: config/express.js [hunk 3]
```diff
@@ -37,8 +31,13 @@ app.use(helmet());
 // enable CORS - Cross Origin Resource Sharing
 app.use(cors());
 
+// eslint-disable-next-line import/newline-after-import
+const swStats = require('swagger-stats');
+app.use(swStats.getMiddleware({}));
+
+//TODO JCB: add express winston to production 
 // enable detailed API logging in dev env
```
Comment (rmharrison) [inline config/express.js:38]: Update comment, or update code?

**Review by rmharrison — CHANGES_REQUESTED** [review]
Please turn-off non-JSON debug logging in production.

File: config/sequelize.js [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 import _ from 'lodash';
 import config from './config';
+import logger from './winston';
 
 let dbLogging;
 if (config.env === 'test') {
     dbLogging = false;
 } else {
-    dbLogging = console.log;
+    dbLogging = (msg) => logger.debug(msg);
```
Comment (rmharrison) [inline config/sequelize.js:10]: Resolves immediate concern around logging raw sequel in PROD.

File: index.js [hunk 5]
```diff
@@ -23,8 +29,8 @@ db.sequelize
   .sync()
   .then(startServer)
   .catch((err) => {
-      if (err) debug('An error occured %j', err);
-      else debug('Database synchronized');
+      if (err) logger.debug('An error occured', err);
```
Comment (rmharrison) [inline index.js:32]: NB: Doesn't look like you've configured winston to error formatting ==> `err = new Error()` object will not actually log.

**Review by rmharrison — APPROVED** [review]
**Blocking item (logging SQL in PROD) resolved in pairing.**

NB: Dev formatter hasn't been updated to show structured log items
```
$ NODE_ENV='development' yarn start

2018-10-09T21:35:09.401Z info: server started on port
```

```
$ NODE_ENV='production' yarn start

{"service":"messaging-service","message":"server started on port","port":4001,"node_env":"production","level":"info"}
```

**Comment by rmharrison** [comment]
* 2-cents / personal preference / git style *
FFR (For Future Reference) / don't chage
I would probably have rebased `branch:logging-updates` to tip of `branch:develop` instead of merging at 881a3d7.

selection_029

**Review by orndorffgrant — APPROVED** [review]
Approved again

**Pull request merged**